### PR TITLE
Add bzk service command

### LINKED ISSUE #163: add a tag option to the bzk run CLI command
Something like:

`bzk run --tag=security`

The tag would then be used to pull the bazooka core images tagged with the same value, e.g. `bazooka/server:security`, etc.

### PR DESCRIPTION
**Pull request by julienvey**
Closes #163 #170 #173
```
Usage: bzk service  COMMAND [arg...]

Manage bazooka service (start, stop, status, upgrade...)

Commands:
  start        Start bazooka
  restart      Restart bazooka
  upgrade      Upgrade bazooka to the latest version
  stop         Stop bazooka
  status       Get bazooka status
```

### CODE REVIEW

File: cli/bzk/service.go [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+	docker "github.com/bywan/go-dockercommand"
+
+	dockerclient "github.com/fsouza/go-dockerclient"
+)
+
+var allContainers []dockerclient.APIContainers
+
+func startService(cmd *cli.Cmd) {
+	cmd.Spec = "[--home|--scm-key|--registry|--docker-sock]... [--tag]"
```
Comment (jawher) [inline cli/bzk/service.go]: `cmd.Spec = "[OPTIONS]"` maybe ?

File: cli/bzk/service.go [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+	config, err := loadConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to load Bazooka config, reason is: %v\n", err)
+	}
+	if len(bzkHome) == 0 {
+		if len(config.Home) == 0 {
+			bzkHome = interactiveInput("Bazooka Home Folder")
+			config.Home = bzkHome
+		} else {
+			bzkHome = config.Home
```
Comment (jawher) [inline cli/bzk/service.go]: Why initialize the `bzkHome` parameter when it is not accessed anymore ?

File: cli/bzk/service.go [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+		}
+	} else {
+		config.Home = bzkHome
+	}
+	if len(dockerSock) == 0 {
+		if len(config.DockerSock) == 0 {
+			dockerSock = interactiveInput("Docker Socket path")
+			config.DockerSock = dockerSock
+		} else {
+			dockerSock = config.DockerSock
```
Comment (jawher) [inline cli/bzk/service.go]: same

File: cli/bzk/service.go [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	} else {
+		config.DockerSock = dockerSock
+	}
+
+	if len(scmKey) == 0 {
+		if len(config.SCMKey) == 0 {
+			scmKey = interactiveInput("Bazooka Default SCM private key")
+			config.SCMKey = scmKey
+		} else {
+			scmKey = config.SCMKey
```
Comment (jawher) [inline cli/bzk/service.go]: same

File: cli/bzk/service.go [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+	} else {
+		config.SCMKey = scmKey
+	}
+
+	if len(tag) == 0 {
+		if len(config.Tag) == 0 {
+			tag = "latest"
+			config.Tag = tag
+		} else {
+			tag = config.Tag
```
Comment (jawher) [inline cli/bzk/service.go]: same

File: cli/bzk/service.go [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+		EnvVar: "BZK_REGISTRY",
+	})
+	dockerSock := cmd.String(cli.StringOpt{
+		Name:   "docker-sock",
+		Desc:   "Location of the Docker unix socket, usually /var/run/docker.sock",
+		EnvVar: "BZK_DOCKERSOCK",
+	})
+	tag := cmd.String(cli.StringOpt{
+		Name: "tag",
+		Desc: "Tag on which run bazooka",
```
Comment (jawher) [inline cli/bzk/service.go]: A more readable description ? Something like "The bazooka version to run" maybe ?

**Comment by julienvey** [comment]
Updated. @jawher can you review again ?

File: cli/bzk/service.go [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		mongoUp := getContainerStatus("bzk_mongodb")
+		serverUp := getContainerStatus("bzk_server")
+		webUp := getContainerStatus("bzk_web")
+
+		if mongoUp && serverUp && webUp {
+			log.Printf("Bazooka service is Up\n")
```
Comment (jawher) [inline cli/bzk/service.go]: Do we need the date/time for this ? wouldn't a simple `fmt.Printf` be better suited ?

**Comment by jawher** [comment]
I just tried the following:

```
$ bzk service stop
$ bzk service start --tag=DEV
```

And I got this message:

> 2015/04/27 20:38:16 Container bzk_mongodb is not `Up`, starting it
> 2015/04/27 20:38:16 API error (404): Cannot start container 11a2efa7b37a8f3397bd14f0f28011f7d471710b965db058e4b5e4142bd2c8e9: [8] System error: stat /mnt/sda1/var/lib/docker/vfs/dir/ca811f4d0e6a13a107b6fb7fa659d320aa76b7a4fabab9cc2b3fea53a80b968b: no such file or directory

I know that there is no bazooka images tagged with DEV, but I wasn't expecting this cryptic error message.

**Comment by jawher** [comment]
After deleting the mongo container and trying to start bazooka again with :

```bzk service start```

I get this error:

> 2015/04/27 20:45:39 Container bzk_mongodb not found, Starting it
> 2015/04/27 20:45:39 Container bzk_server found, but not in the right version, restarting it
> 2015/04/27 20:45:39 Restarting Container bzk_server
> Pulling repository bazooka/server
> 2015/04/27 20:45:45 Tag DEV not found in repository bazooka/server

It is trying to use the DEV tag, but I didn't pass any tag to the start command.

**Comment by jawher** [comment]
LGTM.

**Pull request merged**